### PR TITLE
Add LPG Presence

### PR DESCRIPTION
**Pull request by TheGamerzs**
image
image
image

### CODE REVIEW

File: websites/L/Lime Pro Gaming/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "name": "The Gamerzs",
+    "id": "730142125181894657"
+  },
+  "url": "limeprogaming.com",
+  "description": {
+    "en": "Here at LIME, we have some of the best gaming equipment and accessories."
+  },
+  "service": "Lime Pro Gaming",
+  "altnames": ["LPG", "Lime | Pro Gaming"],
+
```
Comment (EncryptedDev) [inline websites/L/Lime Pro Gaming/dist/metadata.json:13]: Empty line

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Beep boop

File: websites/L/Lime Pro Gaming/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "author": {
+    "name": "The Gamerzs",
+    "id": "730142125181894657"
+  },
+  "url": "limeprogaming.com",
+  "description": {
+    "en": "Here at LIME, we have some of the best gaming equipment and accessories."
+  },
+  "service": "Lime Pro Gaming",
+  "altnames": ["LPG", "Lime | Pro Gaming"],
```
Comment (Bas950) [inline websites/L/Lime Pro Gaming/dist/metadata.json:12]:
```suggestion
  "altnames": ["LPG"],
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — DISMISSED** [review]

**Comment by ririxidev** [comment]
ngl but, I don't really think that some x shop needs presence. sorry but rejecting.

File: websites/L/Lime Pro Gaming/presence.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+let productName, productBrand, blogTitle, blogAuthor;
+const browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+
+  presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:13]:
```suggestion
    largeImageKey: "logo",
    startTimestamp: browsingStamp
  };
```

File: websites/L/Lime Pro Gaming/presence.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
+const presence = new Presence({
+  clientId: "863173597941727282"
+});
+
+let productName, productBrand, blogTitle, blogAuthor;
+const browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:6]:
```suggestion
const presence = new Presence({
  clientId: "863173597941727282"
}), browsingStamp = Math.floor(Date.now() / 1000);

let productName, productBrand, blogTitle, blogAuthor;
```

File: websites/L/Lime Pro Gaming/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "description": {
+    "en": "Here at LIME, we have some of the best gaming equipment and accessories."
+  },
+  "service": "Lime Pro Gaming",
+  "altnames": ["LPG"],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/IZlOaJU.png",
+  "thumbnail": "https://i.imgur.com/FUBK77b.png",
+  "color": "#9bda45",
+  "tags": ["gaming", "retail", "drinks", "distributor", "lpg"],
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/dist/metadata.json:17]:
```suggestion
  "tags": ["gaming", "retail", "drinks", "distributor"],
```

File: websites/L/Lime Pro Gaming/presence.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      )?.textContent || "Viewing News";
+    blogAuthor =
+      document.querySelector(
+        "#shopify-section-article-template > div:nth-child(1) > div > div > div > div > div.tt-autor"
+      )?.textContent || null;
+
+    presenceData.details = blogTitle;
+    presenceData.state = blogAuthor;
+
+    presenceData.smallImageKey = blogAuthor ? "reading" : null;
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:119]: Hmm. No `null` values.
```suggestion
    if(blogAuthor) presenceData.smallImageKey = "reading"
```

File: websites/L/Lime Pro Gaming/presence.ts [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      document.querySelector(
+        "#shopify-section-article-template > div:nth-child(1) > div > div > div > div > h1"
+      )?.textContent || "Viewing News";
+    blogAuthor =
+      document.querySelector(
+        "#shopify-section-article-template > div:nth-child(1) > div > div > div > div > div.tt-autor"
+      )?.textContent || null;
+
+    presenceData.details = blogTitle;
+    presenceData.state = blogAuthor;
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:117]:
```suggestion
    if(blogAuthor) presenceData.state = blogAuthor;
```

File: websites/L/Lime Pro Gaming/presence.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        "#same_product_height > div.tt-breadcrumb > div > ul > li:nth-child(2) > a"
+      ).textContent;
+
+      presenceData.details = productName;
+      presenceData.state = productBrand;
+
+      presenceData.buttons = [
+        { label: "View Product", url: document.location.href },
+        {
+          label: "View Brand/Category",
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:33]:
```suggestion
          label: "View Brand",
```

File: websites/L/Lime Pro Gaming/presence.ts [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+    presenceData.buttons = [
+      {
+        label: "View Page",
+        url: document.location.href
+      }
+    ];
+  } else if (window.location.pathname.includes("/search")) {
+    presenceData.details = `Searching: ${
+      window.location.search.replace("?q=", "").split("&")[0]
```
Comment (ririxidev) [inline websites/L/Lime Pro Gaming/presence.ts:88]: Use array deconstructing.

**Review by rxri — CHANGES_REQUESTED** [review]
Use prettier again and fix things from my review.

**Comment by TheGamerzs** [comment]
Will prettier later

**Review by Slowlife01 — APPROVED** [review]

**Comment by TheGamerzs** [comment]
@Bas950 slowlife suggests that prettier is not needed, so if you can approve or suggest what exactly needs prettier

**Review by Bas950 — APPROVED** [review]

**Pull request merged**